### PR TITLE
Fix nested extension types

### PR DESCRIPTION
**Pull request by jondo2010**
This PR fixes various issues with Extension types at various nesting levels. Rebased onto `main`.

### CODE REVIEW

File: src/array/list/mod.rs [hunk 1]
```diff
@@ -54,7 +54,7 @@ impl<O: Offset> ListArray<O> {
 
         let child_data_type = Self::try_get_child(&data_type)?.data_type();
         let values_data_type = values.data_type();
-        if child_data_type != values_data_type {
+        if child_data_type.to_logical_type() != values_data_type.to_logical_type() {
```
Comment (jorgecarleitao) [inline src/array/list/mod.rs:57]: don't we want this to remain? I.e. if the datatype is `List<Extension(Int32)>` but the array is `List<Int32>`, we want to error, right?

**Review by jorgecarleitao — APPROVED** [review]
Thank you for this PR! I have a minor comment, but otherwise we can merge it.

File: src/array/list/mod.rs (hunk 1 above)
Comment (jondo2010) [reply src/array/list/mod.rs]: Yep, I missed the actual root-cause of the problem, which was in `GrowableStruct`.

**Comment by jondo2010** [comment]
Seems like a codecov submission failure. @jorgecarleitao, can you re-trigger the build on this please?

**Pull request merged**